sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadIT.java

```diff
@@ -61,4 +62,35 @@ public void testReadPublicData() throws Exception {
       // noop
     }
   }
+
+  @Test
+  public void testReadPubsubMessageId() throws Exception {
+    // The pipeline will never terminate on its own
+    pipeline.getOptions().as(DirectOptions.class).setBlockOnRun(false);
+
+    PCollection<PubsubMessage> messages =
+        pipeline.apply(
+            PubsubIO.readMessagesWithMessageId()
+                .fromTopic("projects/pubsub-public-data/topics/taxirides-realtime"));
+
+    messages.apply(
+        "isMessageIdNonNull",
+        signal.signalSuccessWhen(
+            messages.getCoder(),
+            pubsubMessages ->
+                pubsubMessages.stream().noneMatch(m -> Strings.isNullOrEmpty(m.getMessageId()))));
```

Comment:
It looks like the new IT doesn't pass:
https://builds.apache.org/job/beam_PostCommit_Java_PR/169/testReport/junit/org.apache.beam.sdk.io.gcp.pubsub/PubsubReadIT/testReadPubsubMessageId/

Reply:
Looks like the IT passed this time. Could you take another look?

Thanks!